src/mongocxx/test/uri.cpp

```diff
@@ -295,6 +313,228 @@ TEST_CASE("uri::wait_queue_timeout_ms()", "[uri]") {
+static void run_srv_max_hosts_test_file(bsoncxx::document::view test_doc) {
+    using namespace mongocxx;
+
+    bool expect_exception = test_doc["error"] && test_doc["error"].get_bool().value;
+    bool should_ping = true;
+    if (test_doc["ping"] && test_doc["ping"].get_bool() == false) {
+        should_ping = false;
+    }
+    should_ping = should_ping && !expect_exception;
+
+    options::apm apm_opts;
+    std::mutex mtx;
+    std::vector<std::string> new_hosts;
+
+    apm_opts.on_topology_changed([&](const events::topology_changed_event& event) {
+        const std::lock_guard<std::mutex> lock(mtx);
+        auto new_td = event.new_description();
+        auto servers = new_td.servers();
+        new_hosts.clear();
+        for (const auto& i : servers) {
+            new_hosts.push_back(std::string(i.host()) + ":" + std::to_string(i.port()));
+        }
+    });
+
+    mongocxx::options::tls tls_options;
+    mongocxx::options::client client_options;
+
+    auto ca_file_path = std::getenv("MONGOCXX_TEST_TLS_CA_FILE");
+    REQUIRE(ca_file_path);
+    tls_options.ca_file(ca_file_path);
+    tls_options.allow_invalid_certificates(true);
+    client_options.tls_opts(tls_options);
+    client_options.apm_opts(apm_opts);
+
+    mongocxx::uri my_uri;
+    try {
+        my_uri = uri{test_doc["uri"].get_string().value};
+        class client client {
+            my_uri, test_util::add_test_server_api(client_options)
+        };
+        bool using_tls = client.uri().tls();
+        if (!using_tls) {
+            return;
+        }
+        REQUIRE(!expect_exception);
+        if (should_ping) {
+            validate_srv_max_hosts(std::move(client), test_doc, mtx, new_hosts);
+        }
+    } catch (mongocxx::exception& e) {
+        REQUIRE(expect_exception);
+    }
+}
+
+static void iterate_srv_max_hosts_tests(std::string dir, std::vector<std::string> files) {
+    for (const auto& file : files) {
+        auto test_doc = _doc_from_file("/initial_dns_seedlist_discovery/" + dir + "/" + file);
+        run_srv_max_hosts_test_file(test_doc);
+    }
+}
+
+static bool is_tls_enabled(void) {
+    using namespace mongocxx;
+
+    options::tls tls_options;
+    options::client client_options;
+
+    auto ca_file_path = std::getenv("MONGOCXX_TEST_TLS_CA_FILE");
+    REQUIRE(ca_file_path);
+    tls_options.ca_file(ca_file_path);
+    tls_options.allow_invalid_certificates(true);
+    client_options.tls_opts(tls_options);
+
+    try {
+        class mongocxx::client client {
+            uri{"mongodb://localhost:27017/?tls=true"},
+                test_util::add_test_server_api(client_options)
+        };
+
+        auto admin = client["admin"];
+        auto result = admin.run_command(make_document(kvp("ping", 1)));
+        REQUIRE(!result.empty());
+        REQUIRE(result.begin()->get_double() == 1.0);
+        return true;
+    } catch (mongocxx::exception& e) {
+        return false;
+    }
+}
+
+TEST_CASE("uri::test_srv_max_hosts", "[uri]") {
```

Comment:
```suggestion
TEST_CASE("uri::test_srv_max_hosts", "[uri]") {
   mongocxx::instance::current();

```
Ensure libmongoc is initialized before executing tests that use libmongoc functions.